R/get_coding_ssm.R

```diff
@@ -0,0 +1,153 @@
+#' @param limit_pathology Supply this to restrict mutations to one pathology.
+#' @param limit_samples Supply this to restrict mutations to a vector of sample_id (instead of subsetting using the provided metadata)
+#' @param these_samples_metadata Supply a metadata table to auto-subset the data to samples in that table before returning
+#' @param force_unmatched_samples Optional argument for forcing unmatched samples, using [GAMBLR.data::get_ssm_by_samples].
+#' @param projection Reference genome build for the coordinates in the MAF file. The default is hg19 genome build.
+#' @param seq_type The seq_type you want back, default is genome.
+#' @param basic_columns Set to FALSE to override the default behavior of returning only the first 45 columns of MAF data.
+#' @param maf_cols if basic_columns is set to FALSE, the user can specify what columns to be returned within the MAF. This parameter can either be a vector of indexes (integer) or a vector of characters (matching columns in MAF).
+#' @param from_flatfile This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param augmented This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param min_read_support Only returns variants with at least this many reads in t_alt_count (for cleaning up augmented MAFs)
+#' @param groups This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param include_silent Logical parameter indicating whether to include silent mutations into coding mutations. Default is TRUE.
+#' @param engine This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#'
+#' @return A data frame (.maf) with coding SSMs in reference to the selected projection.
+#' 
+#' @export
+#' 
+#' @examples
+#' #return SSMs in reference to GRCh37:
+#' ssm_grch37 = get_coding_ssm(seq_type = "genome")
+#'
+#' #return SSMs in reference to hg38:
+#' dohh2_meta = GAMBLR.data::sample_data$meta %>% dplyr::filter(sample_id == "DOHH-2")
+#' ssm_hg38 = get_coding_ssm(projection = "hg38", these_samples_metadata = dohh2_meta, seq_type = "genome")
+#' 
+get_coding_ssm = function(limit_cohort,
+                          exclude_cohort,
+                          limit_pathology,
+                          limit_samples,
+                          these_samples_metadata,
+                          force_unmatched_samples,
+                          projection = "grch37",
+                          seq_type,
+                          basic_columns = TRUE,
+                          maf_cols = NULL,
+                          from_flatfile = NULL,
+                          augmented = NULL,
+                          min_read_support = 3,
+                          groups = NULL,
+                          include_silent = TRUE,
```

Comment:
```
ssm_grch37 = get_coding_ssm(seq_type = "genome",include_silent = F)
Using the bundled SSM calls (.maf) calls in GAMBLR.data...
Error in get_coding_ssm(seq_type = "genome", include_silent = F) : 
  object 'coding_class' not found
```

Reply:
The fix to this relates to the issue I noted in the comment above.

Reply:
Is this the same global variable in GAMBLR.helpers ([this](https://github.com/morinlab/GAMBLR.helpers/blob/main/R/helpers-vars.R) file)? It could be imported with `usethis::use_import_from("GAMBLR.helpers", "coding_class")` to avoid duplications.

Reply:
Yes, the most recent commit actually removes this global variable from this package and instead calls the same variable from GAMBLR.helpers.

Reply:
I see the `coding_class` creation in GAMBLR.data was removed, but I think its import from GAMBLR.helpers is still missing. 
```
  Error in assign_cn_to_ssm(this_sample_id = "DOHH-2", coding_only = TRUE) : 
    object 'coding_class' not found
```

Reply:
GAMBLR.helpers already needs GAMBLR.data. After running `usethis::use_import_from("GAMBLR.helpers", "coding_class")`, I got this: 
```
Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) : 
  cyclic namespace dependency detected when loading ‘GAMBLR.data’, already loading ‘GAMBLR.helpers’, ‘GAMBLR.data’
```

Reply:
`coding_class` has been added back to this pcakge.